frontend/src/components/form/searchbar/searchbar.component.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { HTMLAttributes, useEffect, useRef, useState } from "react";
+import { IonItem, IonList, IonSearchbar } from "@ionic/react";
+import { Item } from "@/types/item.ts";
+import './searchbar.component.css';
+
+interface Props {
+  values: Array<Item>;
+  onValueSelected: (value: Item) => void;
+}
+
+export const Searchbar: React.FC<Props & HTMLAttributes<HTMLIonSearchbarElement>> = (props) => {
+  const searchbarRef = useRef<HTMLIonSearchbarElement | null>(null);
+
+  const [search, setSearch] = useState<string | null>();
+  const [searchResult, setSearchResult] = useState<Array<any>>([]);
+
+  useEffect(() => {
+    if (!search) return setSearchResult([]);
+    const searchData = search.split(' ').filter(s => s).map(s => s.toLowerCase());
+    setSearchResult(
+      props.values.filter(value => {
+        const valueData = value.label.split(' ').filter(v => v).map(v => v.toLowerCase());
+        for (const s of searchData) {
+          if (valueData.find(v => v.includes(s))) continue;
+          return false;
+        }
+        return true;
```

Comment:
Not sure to understand what this part is doing? Won't it be easier to use `searchData.every()` of `searchData.some()`?

Reply:
Indeed! I didn't know these functions, it's way better, thank you for your help :)